virtcontainers/acrn_arch_base.go

```diff
@@ -0,0 +1,762 @@
+	kernelParamsNonDebug []Param
+	kernelParamsDebug    []Param
+	kernelParams         []Param
+}
+
+const acrnPath = "/usr/bin/acrn-dm"
+const acrnctlPath = "/usr/bin/acrnctl"
+
+var baselogger *logrus.Entry
+
+// AcrnBlkDevPoolSz defines the number of dummy Blk
+// that will be created for hot-plugging virtio-blk
+// device.
+var AcrnBlkDevPoolSz = 8
+
+// AcrnBlkdDevSlot array provides translation between
+// the vitio-blk device index and slot it is currently
+// attached.
+// Allocating extra 1 to accommodate for VM rootfs
+// which is at driveIndex 0
+var AcrnBlkdDevSlot = make([]int, AcrnBlkDevPoolSz+1)
+
+// acrnKernelParamsNonDebug is a list of the default kernel
+// parameters that will be used in standard (non-debug) mode.
+var acrnKernelParamsNonDebug = []Param{
+	{"quiet", ""},
+}
+
+// acrnKernelParamsSystemdNonDebug is a list of the default systemd related
+// kernel parameters that will be used in standard (non-debug) mode.
+var acrnKernelParamsSystemdNonDebug = []Param{
+	{"systemd.show_status", "false"},
+}
+
+// acrnKernelParamsDebug is a list of the default kernel
+// parameters that will be used in debug mode (as much boot output as
+// possible).
+var acrnKernelParamsDebug = []Param{
+	{"debug", ""},
+}
+
+// acrnKernelParamsSystemdDebug is a list of the default systemd related kernel
+// parameters that will be used in debug mode (as much boot output as
+// possible).
+var acrnKernelParamsSystemdDebug = []Param{
+	{"systemd.show_status", "true"},
+	{"systemd.log_level", "debug"},
+	{"systemd.log_target", "kmsg"},
+	{"printk.devkmsg", "on"},
+}
+
+var acrnKernelRootParams = []Param{
+	{"root", "/dev/vda1 rw rootwait"},
+}
+
+var acrnKernelParams = []Param{
+	{"tsc", "reliable"},
+	{"no_timer_check", ""},
+	{"nohpet", ""},
+	{"console", "tty0"},
+	{"console", "ttyS0"},
+	{"console", "hvc0"},
+	{"log_buf_len", "16M"},
+	{"consoleblank", "0"},
+	{"iommu", "off"},
+	{"i915.avail_planes_per_pipe", "0x070F00"}, //this can also be passed via configuration.toml
+	{"i915.enable_hangcheck", "0"},
+	{"i915.nuclear_pageflip", "1"},
+	{"i915.enable_guc_loading", "0"},
+	{"i915.enable_guc_submission", "0"},
+	{"i915.enable_guc", "0"},
+}
+
+// Device is the acrn device interface.
+type Device interface {
+	Valid() bool
+	AcrnParams(slot int, config *Config) []string
+}
+
+// ConsoleDeviceBackend is the character device backend for acrn
+type ConsoleDeviceBackend string
+
+const (
+
+	// Socket creates a 2 way stream socket (TCP or Unix).
+	Socket ConsoleDeviceBackend = "socket"
+
+	// Stdio sends traffic from the guest to ACRN's standard output.
+	Stdio ConsoleDeviceBackend = "console"
+
+	// File backend only supports console output to a file (no input).
+	File ConsoleDeviceBackend = "file"
+
+	// TTY is an alias for Serial.
+	TTY ConsoleDeviceBackend = "tty"
+
+	// PTY creates a new pseudo-terminal on the host and connect to it.
+	PTY ConsoleDeviceBackend = "pty"
+)
+
+// BEPortType marks the port as console port or virtio-serial port
```

Comment:
OOI, what does "BE" stand for? Back end?

Reply:
yes backend.